app/models/order.rb

```diff
@@ -429,13 +431,35 @@ def self.orders_count_per_state(orders)
     orders.each { |key, value| orders[key] = value.count }
   end
 
-  def self.generate_gc_code
-    record = where(detail_type: "GoodCity").order("id desc").first
-    "GC-" + gc_code(record).to_s.rjust(5, "0")
+  def self.order_code_prefix(detail_type)
```

Comment:
This is good. Can you put it in a concern? This code has a single responsibility and would fit neatly inside a concern where all relevant methods can be kept together.